### ISSUE TITLE
Rare Fatal NSInvalidArgumentException Thrown from VideoPlayerViewController.reset

### ISSUE THREAD

**Issue by bcmn** [insidegui/WWDC#208]
Description: "An instance of AVPlayer cannot remove a time observer that was added by a different instance of AVPlayer."

Apparent race condition.

**Comment by insidegui** [comment]
Fabric link: https://fabric.io/canyon/mac/apps/io.wwdc.app/issues/592f1bafbe077a4dcc7e7efb?time=last-seven-days

Stack trace:

```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x7fff7a73a2b3 (Missing)
1  libobjc.A.dylib                0x7fff8f54548d (Missing)
2  AVFoundation                   0x7fff76e61d3a (Missing)
3  WWDC                           0x10098d709 specialized VideoPlayerViewController.reset(oldValue : AVPlayer?) -> () (VideoPlayerViewController.swift)
4  WWDC                           0x10096345b specialized AppCoordinator.shelfViewControllerDidSelectPlay(ShelfViewController) -> () (AppCoordinator+Shelf.swift)
5  WWDC                           0x100962950 protocol witness for ShelfViewControllerDelegate.shelfViewControllerDidSelectPlay(ShelfViewController) -> () in conformance AppCoordinator (AppCoordinator+Shelf.swift)
6  WWDC                           0x1009115ba @objc ShelfViewController.play(Any?) -> () (ShelfViewController.swift)
7  libsystem_trace.dylib          0x7fff9005c3a7 (Missing)
8  AppKit                         0x7fff78928721 (Missing)
9  WWDC                           0x1009719d7 specialized VibrantButton.mouseUp(with : NSEvent) -> () (VibrantButton.swift:101)
10 WWDC                           0x100971156 @objc VibrantButton.mouseDown(with : NSEvent) -> ()
11 AppKit                         0x7fff783297a9 (Missing)
12 AppKit                         0x7fff78aa08de (Missing)
13 AppKit                         0x7fff78a9ff0a (Missing)
14 AppKit                         0x7fff78924681 (Missing)
15 AppKit                         0x7fff7819f427 (Missing)
16 AppKit                         0x7fff78169e0e (Missing)
17 libdyld.dylib                  0x7fff8fe2a235 (Missing)
18 ???                            0x2 (Missing)
```

**Comment by insidegui** [comment]
Fixed in #210
